worker/src/response.rs

```diff
@@ -107,6 +107,27 @@ impl Response {
         })
     }
 
+    /// Create a `Response` which redirects to the specified URL with default status_code of 302
+    pub fn redirect(url: url::Url) -> Result<Self> {
+        match EdgeResponse::redirect(&url.as_str()) {
+            Ok(edge_response) => Ok(Response::from(edge_response)),
+            Err(err) => Err(Error::from(err)),
+        }
+    }
+
+    /// Create a `Response` which redirects to the specified URL with a custom status_code
+    pub fn redirect_with_status_code(url: url::Url, status_code: u16) -> Result<Self> {
+        if !(300..=399).contains(&status_code) {
+            return Err(Error::Internal(
+                "provided error status code is invalid".into(),
```

Comment:
super minor nit: could this be a little more specific and say something like `"redirect status codes must be in the 300 range! see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#redirection_messages for more"`?

Reply:
This might be the same error string used in another "invalid status" scenario in `Response::error`, in which case we should probably change both to be more descriptive. Adding the specific range to both errors would be great.

Reply:
can we also change the name to `redirect_with_status` ? thank you!